wandb/sdk_py27/wandb_run.py

```diff
@@ -769,6 +771,67 @@ def entity(self):
         """
         return self._entity or ""
 
+    def _label(
+        self,
+        code = None,
+        repo = None,
+        code_version = None,
+        **kwargs
+    ):
+        if self._settings.label_disable:
+            return
+        with telemetry.context(run=self) as tel:
+            if code and RE_LABEL.match(code):
+                tel.label.code_string = code
+            if repo and RE_LABEL.match(repo):
+                tel.label.repo_string = repo
+            if code_version and RE_LABEL.match(code_version):
+                tel.label.code_version = code_version
+
+    def _label_probe_lines(self, lines):
+        if not lines:
+            return
+        parsed = telemetry._parse_label_lines(lines)
+        if not parsed:
+            return
+        label_dict = {}
+        code = parsed.get("code") or parsed.get("c")
+        if code:
+            label_dict["code"] = code
+        repo = parsed.get("repo") or parsed.get("r")
+        if repo:
+            label_dict["repo"] = repo
+        code_ver = parsed.get("version") or parsed.get("v")
+        if code_ver:
+            label_dict["code_version"] = code_ver
+        self._label(**label_dict)
+
+    def _label_probe_main(self):
+        m = sys.modules.get("__main__")
+        if not m:
+            return
+        doc = getattr(m, "__doc__", None)
+        if not doc:
+            return
+
+        doclines = doc.splitlines()
+        self._label_probe_lines(doclines)
+
+    # TODO: annotate jupyter Notebook class
+    def _label_probe_notebook(self, notebook):
+        logger.info("probe notebook")
+        lines = None
+        try:
+            data = notebook.probe_ipynb()
+            cell0 = data.get("cells", [])[0]
+            lines = cell0.get("source")
+        except Exception as e:
+            print("ERR", e)
```

Comment:
```suggestion
            termerror(e)
```

Reply:
thanks. this is supposed to be silent, so just the logger is good enough. fixd